Conch/lib/Conch/Controller/DatacenterRackRole.pm

```diff
@@ -0,0 +1,127 @@
+=pod
+
+=head1 NAME
+
+Conch::Controller::DatacenterRackRole
+
+=head1 METHODS
+
+=cut
+
+package Conch::Controller::DatacenterRackRole;
+
+use Mojo::Base 'Mojolicious::Controller', -signatures;
+use Conch::UUID 'is_uuid';
+
+use Conch::Models;
+
+
+=head2 under
+
+Supports rack role lookups by uuid and name
+
+=cut
+
+sub under ($c) {
+	my $r;
+
+	if($c->param('id') =~ /^(.+?)\=(.+)$/) {
+		if($1 eq 'name') {
+			$r = Conch::Model::DatacenterRackRole->from_name($2);
+		} else {
+			$c->status(404 => { error => "Not found" });
+			return undef;
+		}
+	} else {
+		$r = Conch::Model::DatacenterRackRole->from_id($c->param('id'));
+	}
+
+	if ($r) {
+		$c->stash('rack_role' => $r);
+		return 1;
+	} else {
+		$c->status(404 => { error => "Not found" });
+		return undef;
+	}
+}
+
+=head2 create
+
+=cut
+
+sub create ($c) {
+	return $c->status(403) unless $c->is_global_admin;
+	my $d = $c->validate_input('RackRoleCreate') or return;
+
+	my $r = Conch::Model::DatacenterRackRole->new($d->%*)->save();
+	$c->status(303);
+	$c->redirect_to($c->url_for("/rack_role/".$r->id));
+
+}
+
+=head2 get
+
+Get a single rack role
+
+=cut
+
+sub get ($c) {
```

Comment:
Likewise, should restrict to global admin I think.

Reply:
Fixed.